types/jest/jest-tests.ts

```diff
@@ -307,10 +307,15 @@ const mock7 = jest.fn((arg: number) => arg);
 const mock8: jest.Mock = jest.fn((arg: number) => arg);
 // $ExpectType Mock<Promise<boolean>, [number, string, {}, [], boolean]>
 const mock9 = jest.fn((a: number, _b: string, _c: {}, _iReallyDontCare: [], _makeItStop: boolean) => Promise.resolve(_makeItStop));
-// $ExpectType Mock<never, [never]>
-const mock10 = jest.fn((arg: never) => { throw new Error(arg); });
+// $ExpectType Mock<never, []>
+const mock10 = jest.fn(() => { throw new Error(); });
 // $ExpectType Mock<unknown, [unknown]>
 const mock11 = jest.fn((arg: unknown) => arg);
+interface TestApi {
+    test(x: number): string;
+}
+// $ExpectType Mock<string, [number]>
+const mock12 = jest.fn<ReturnType<TestApi["test"]>, ArgsType<TestApi["test"]>>();
```

Comment:
Just a note: the ArgsType reference will break after https://github.com/DefinitelyTyped/DefinitelyTyped/pull/32816 is merged

Reply:
It doesn't seem to be a problem as the only change was that ArgsType was moved inside the jest namespace. I'm referencing it within the same namespace so it doesn't need the jest. prefix (tested).

Reply:
@jwbay you were right. This fails in master. I fixed it in #33117